src/codegeneration/InstantiateModuleTransformer.js

```diff
@@ -82,7 +83,15 @@ class DeclarationExtractionTransformer extends HoistVariablesTransformer {
     this.declarations_ = [];
   }
   getDeclarationStatements() {
-    return [this.getVariableStatement(), ...this.declarations_];
+    let statement = this.getVariableStatement();
+    let statements;
+    if (statement.type === ANON_BLOCK) {
```

Comment:
I think a comment here would be helpful. Is another case where we are downstream of a transformation rather than operating on normal code?  In any case it's not obvious to me why we lift the statements here.